### PR TITLE
Pages delete entry

### LINKED ISSUE #14: [pages] Ability to delete existing entries

### PR DESCRIPTION
**Pull request by heafalan**
Fixes #14

### CODE REVIEW

File: packages/pages/src/App.js [hunk 1]
```diff
@@ -41,6 +41,21 @@ const App = ({ effects, state }) => (
...
+          id="idEntry"
+          name="id"
+          value={state.id}
+          onChange={effects.changeId}
+        />
+        <button type="button" onClick={effects.deleteEntry}>
+          Delete
+        </button>
+      </label>
+    </div>
```
Comment (julien-f) [inline packages/pages/src/App.js]: IMHO, a button next to an entries to delete it would be easier to use.

File: packages/pages/src/App.js [hunk 2]
```diff
@@ -70,6 +86,16 @@ export default provideState({
       this.state.content = "";
       await this.effects.refreshEntries();
     },
+    async deleteEntry() {
+      await fetch("/api/", {
+        method: "post",
+        body: format.request(0, "deleteEntry", {
+          id: this.state.id,
+        }),
+      });
```
Comment (julien-f) [inline packages/pages/src/App.js:87]: You need to parse and check the response to be sure the entry has been correctly deleted.

File: packages/pages/src/App.js [hunk 3]
```diff
@@ -70,6 +72,28 @@ export default provideState({
...
+      const response = await fetch("/api/", {
+        method: "post",
+        body: format.request(0, "deleteEntry", {
+          id: this.state.id,
+        }),
+      });
+      this.state.id = "";
+      const parsed = parse(await response.text());
+      if(parsed.type === 'response') {
+        console.log(parsed.result);
```
Comment (julien-f) [inline packages/pages/src/App.js]: Remove this.

File: packages/pages/src/App.js [hunk 4]
```diff
@@ -70,6 +72,28 @@ export default provideState({
...
+        await this.effects.refreshEntries();
+      } else if (parsed.type === 'error') {
+        console.log(parsed.error);
```
Comment (julien-f) [inline packages/pages/src/App.js]: Good idea to log an error, use `console.warn` instead.

File: packages/pages/src/App.js [hunk 5]
```diff
@@ -32,7 +33,7 @@ const App = ({ effects, state }) => (
         <input
           type="text"
           id="contentEntry"
-          name="name"
+          name="content"
```
Comment (julien-f) [inline packages/pages/src/App.js]: Why this change?

File: packages/pages/src/App.js [hunk 6]
```diff
@@ -70,6 +72,28 @@ export default provideState({
       this.state.content = "";
       await this.effects.refreshEntries();
     },
+    async deleteEntry(
+      _,
+      {
+        target: { value },
+      }
+    ) {
+      this.state.id = value;
```
Comment (julien-f) [inline packages/pages/src/App.js]: Why do you need this?

File: packages/pages/src/App.js (hunk 5 above)
Comment (heafalan) [reply packages/pages/src/App.js]: because the item in question is the "content".

File: packages/pages/src/App.js (hunk 6 above)
Comment (heafalan) [reply packages/pages/src/App.js]: This allows to recover the "value".

File: packages/pages/src/App.js (hunk 5 above)
Comment (julien-f) [reply packages/pages/src/App.js]: Ok, but I don't think we are using these `name` attributes.

If we are not, I think we should remove them.

File: packages/pages/src/App.js (hunk 6 above)
Comment (julien-f) [reply packages/pages/src/App.js]: I don't understand what you mean, you can use `value` directly in the `fetch` call.

File: packages/pages/src/App.js [hunk 7]
```diff
@@ -70,6 +80,26 @@ export default provideState({
...
+      const response = await fetch("/api/", {
+        method: "post",
+        body: format.request(0, "deleteEntry", {
+          id: this.state.id,
+        }),
+      });
+      this.state.id = "";
+      await this.effects.refreshEntries();
```
Comment (julien-f) [inline packages/pages/src/App.js]: Only refresh in case of success.

File: packages/pages/src/App.js [hunk 8]
```diff
@@ -12,6 +12,15 @@ const App = ({ effects, state }) => (
         {state.entries.map(entry => (
           <li key={entry.id}>
             {entry.name} {entry.content}
+            <button
+              type="button"
+              id="idEntry"
```
Comment (julien-f) [inline packages/pages/src/App.js]: Why do you need this attribute?

File: packages/pages/src/App.js [hunk 9]
```diff
@@ -12,6 +12,15 @@ const App = ({ effects, state }) => (
...
+              name="deleteEntry"
```
Comment (julien-f) [inline packages/pages/src/App.js]: Why do you need this attribute?

File: packages/pages/src/App.js [hunk 10]
```diff
@@ -21,8 +28,6 @@ const App = ({ effects, state }) => (
         Name
         <input
           type="text"
-          id="nameEntry"
-          name="name"
```
Comment (julien-f) [inline packages/pages/src/App.js:25]: This has nothing to do with this PR, I'll merge it but next time it should go in its own PR.

**Review by julien-f — APPROVED** [review]
